### PR TITLE
Expand histogram gallery example to show usage of pattern as fill

### PR DESCRIPTION
**Pull request by michaelgrund**
**Description of proposed changes**

Adresses #2323 and expands the histogram example to show the use of a pattern as fill.

**Preview:** https://pygmt-dev--2421.org.readthedocs.build/en/2421/gallery/histograms/histogram.html

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by seisman** [comment]
I'm OK with the changes, but I feel it would be better if we can generate two different random datasets and plot two overlapping histograms, just like @JingHuiTong's example in:
image
The example also shows how to add legends for multiple histograms.

**Comment by michaelgrund** [comment]
> I'm OK with the changes, but I feel it would be better if we can generate two different random datasets and plot two overlapping histograms, just like @JingHuiTong's example in [#2323 (comment)](https://github.com/GenericMappingTools/pygmt/issues/2323#issuecomment-1383567307): image The example also shows how to add legends for multiple histograms.

I'm fine with that changes although I'm not a big fan of overlapping histograms (even if pattern and color is used to distinguish them). What do others think @GenericMappingTools/pygmt-maintainers ?

File: examples/gallery/histograms/histogram.py [hunk 1]
```diff
@@ -4,7 +4,8 @@
 The :meth:`pygmt.Figure.histogram` method can plot regular histograms.
 Using the ``series`` parameter allows to set the interval for the width of
 each bar. The type of the histogram (frequency count or percentage) can be
-selected via the ``histtype`` parameter.
+selected via the ``histtype`` parameter. The fill can be either a
+standard color name or a pattern.
```
Comment (yvonnefroehlich) [inline examples/gallery/histograms/histogram.py:8]:
```suggestion
selected via the ``histtype`` parameter. The ``fill`` of the
bars can be either a standard color name or a pattern.
```

File: examples/gallery/histograms/histogram.py [hunk 2]
```diff
@@ -19,19 +20,39 @@
...
+            data=data,
+            # Define the frame, add title and set background color to
+            # "lightgray", add labels for x and y axes
+            frame=["WSne+glightgray", "x+lElevation in m", "y+lCounts"],
+            # Generate evenly spaced bins by increments of 5
+            series=5,
+            # Use "red3" as color fill for the bars
+            fill="red3",
+            # Use a pen size of 1 point to draw the outlines
+            pen="1p",
```
Comment (yvonnefroehlich) [inline examples/gallery/histograms/histogram.py:40]:
```suggestion
            # Use a pen width of 1 point to draw the outlines
            pen="1p",
```

File: examples/gallery/histograms/histogram.py [hunk 3]
```diff
@@ -4,7 +4,8 @@
 The :meth:`pygmt.Figure.histogram` method can plot regular histograms.
 Using the ``series`` parameter allows to set the interval for the width of
 each bar. The type of the histogram (frequency count or percentage) can be
-selected via the ``histtype`` parameter.
+selected via the ``histtype`` parameter. The ``fill`` of the
+bars can be either a standard color name or a pattern.
```
Comment (yvonnefroehlich) [inline examples/gallery/histograms/histogram.py:8]: Maybe we should simply write "color" as beside one of the GMT standard color names users can pass a string for R/G/B levels (e.g., `"100/123/52"`) or hex value (e.g., `"#333333"`) even we are not showing this explicitly in this gallery example?

```suggestion
selected via the ``histtype`` parameter. The ``fill`` of the
bars can be either a color or a pattern.
```

**Comment by seisman** [comment]
> > I'm OK with the changes, but I feel it would be better if we can generate two different random datasets and plot two overlapping histograms, just like @JingHuiTong's example in [#2323 (comment)](https://github.com/GenericMappingTools/pygmt/issues/2323#issuecomment-1383567307): image The example also shows how to add legends for multiple histograms.
> 
> I'm fine with that changes although I'm not a big fan of overlapping histograms (even if pattern and color is used to distinguish them). What do others think @GenericMappingTools/pygmt-maintainers ?



> > I'm OK with the changes, but I feel it would be better if we can generate two different random datasets and plot two overlapping histograms, just like @JingHuiTong's example in [#2323 (comment)](https://github.com/GenericMappingTools/pygmt/issues/2323#issuecomment-1383567307): image The example also shows how to add legends for multiple histograms.
> 
> I'm fine with that changes although I'm not a big fan of overlapping histograms (even if pattern and color is used to distinguish them). What do others think @GenericMappingTools/pygmt-maintainers ?

Overlapping histograms are sometimes necessary and useful (e.g., comparing earthquake depth distributions of an initial catalog and a relocated catalog). But I agree that a gallery example for a simple histogram is still necessary. 

What about keeping the current gallery example unchanged (i.e., closing this PR without merging. Sorry for that.) and having another gallery example for overlapping histograms filled with patterns?

**Comment by yvonnefroehlich** [comment]
> > > I'm OK with the changes, but I feel it would be better if we can generate two different random datasets and plot two overlapping histograms, just like @JingHuiTong's example in [#2323 (comment)](https://github.com/GenericMappingTools/pygmt/issues/2323#issuecomment-1383567307). The example also shows how to add legends for multiple histograms.
> > 
> > 
> > I'm fine with that changes although I'm not a big fan of overlapping histograms (even if pattern and color is used to distinguish them). What do others think @GenericMappingTools/pygmt-maintainers ?
> 
> Overlapping histograms are sometimes necessary and useful (e.g., comparing earthquake depth distributions of an initial catalog and a relocated catalog). But I agree that a gallery example for a simple histogram is still necessary.
> 
> What about keeping the current gallery example unchanged (i.e., closing this PR without merging. Sorry for that.) and having another gallery example for overlapping histograms filled with patterns?

In general, I prefer non-overlapping histograms. But probably it is fair to show how this histogram type can be generated.

Maybe a tutorial on histograms beside this gallery example is a solution herer? In a tutorial, we can show the different fill options for the bars, but also how to create cumulative, overlapping, grouped, stacked, horizontal etc. histograms (here as subplot to not post several figues). Maybe we can also state limitations or concerns regarding this kind of visualizations?
histo_versions


<details><summary>click to show the code</summary>
<p>

```python

import numpy as np
import pygmt

np.random.seed(100)

# Generate random elevation data from a normal distribution
mean = 100  # mean of distribution
stddev = 20  # standard deviation of distribution
data01 = mean + stddev * np.random.randn(50)
data02 = mean + stddev*2 * np.random.randn(50)
data_merge = np.concatenate((data01, data02), axis=None)

R_histo = [0, 300, 0, 51]
bin_histo = 10

fig = pygmt.Figure()

with fig.subplot(
    nrows=2,
    ncols=3,
    figsize=("20c", "10c"),
    title="Histograms",
    frame=["WSne", "xaf10", "ya10"],
    sharex="b+lElevation in m",
    sharey="l+lConunts",
):
    
    with fig.set_panel(panel=0):
        fig.histogram(
            data=data01,
            region=R_histo,
            series=bin_histo,
            fill="red3",
            pen="1p",
            histtype=0,
            label="data01"
        )
        fig.legend()

    with fig.set_panel(panel=1):
        fig.histogram(
            data=data02,
            region=R_histo,
            series=bin_histo,
            fill="orange",
            pen="1p",
            histtype=0,
            label="data02",
        )
        fig.legend()

    with fig.set_panel(panel=2):
        fig.histogram(
            data=data01,
            region=R_histo,
            series=bin_histo,
            fill="red3",
            pen="1p",
            histtype=0,
        )
        fig.histogram(
            data=data02,
            region=R_histo,
            series=bin_histo,
            fill="orange@50",
            pen="1p",
            histtype=0,
            label="overlapping",
        )
        fig.legend()

    with fig.set_panel(panel=3):
        fig.histogram(
            data=data01,
            region=R_histo,
            series=bin_histo,
            fill="red3",
            pen="1p",
            histtype=0,
            label="data01"
        )
        fig.histogram(
            data=data01,
            region=R_histo,
            series=bin_histo,
            fill="p8+b",
            pen="1p",
            histtype=0,
            cumulative=True,
            label="cumulative"
        )
        fig.legend()

    with fig.set_panel(panel=4):
        fig.histogram(
            data=data_merge,
            region=R_histo,
            series=bin_histo,
            fill="orange",
            pen="1p",
            histtype=0,
            label="stacked",
        )
        fig.histogram(
            data=data01,
            region=R_histo,
            series=bin_histo,
            fill="red3",
            pen="1p",
            histtype=0,
        )
        fig.legend()

    with fig.set_panel(panel=5):
        fig.histogram(
            data=data01,
            region=R_histo,
            series=bin_histo,
            fill="red3",
            pen="1p",
            histtype=0,
            barwidth=str(bin_histo/2) + "+o-" + str(bin_histo/4),
        )
        fig.histogram(
            data=data02,
            region=R_histo,
            series=bin_histo,
            fill="orange",
            pen="1p",
            histtype=0,
            barwidth=str(bin_histo/2) + "+o" + str(bin_histo/4),
            label="grouped",
        )
        fig.legend()
    
fig.show()
# fig.savefig(fname="histo_versions.png")
```
</p>
</details>


I am fine with not expanding this gallery example. But in case we still want to do so, would this be an option? I feel here the overlapping is visually not misleading.
_Code example_: (so far, only code, no comments)
```python
import numpy as np
import pygmt

np.random.seed(100)

# Generate random elevation data from a normal distribution
mean = 100  # mean of distribution
stddev = 20  # standard deviation of distribution
data = mean + stddev * np.random.randn(50)

region_histo = [0, 200, 0, 51]
bin_histo = 10

fig = pygmt.Figure()

with fig.subplot(
    nrows=1,
    ncols=2,
    figsize=("20c", "10c"),
    title="Histograms",
    frame=["WSne", "xaf10", "ya10"],
    sharex="b+lElevation in m",
    sharey="l+lConunts",
):
    
    with fig.set_panel(panel=0):
        fig.histogram(
            data=data,
            region=region_histo,
            series=bin_histo,
            fill="red3",
            pen="1p",
            histtype=0,
        )

    with fig.set_panel(panel=1):
        fig.histogram(
            data=data,
            region=region_histo,
            series=bin_histo,
            fill="red3",
            pen="1p",
            histtype=0,
            label="data",
        )
        fig.histogram(
            data=data,
            region=region_histo,
            series=bin_histo,
            fill="p8+b",
            pen="1p",
            histtype=0,
            cumulative=True,
            label="data cumulative",
        )
        fig.legend(position="jLT")
    
fig.show()
# fig.savefig(fname="histo_cumulative.png")
```
_Output figure_:
histo_cumulative

File: examples/gallery/histograms/histogram.py [hunk 4]
```diff
@@ -19,19 +20,39 @@
...
-    # Use the pen parameter to draw the outlines with a width of 1 point
-    pen="1p",
-    # Choose histogram type 0, i.e., counts [Default]
-    histtype=0,
-)
+with fig.subplot(
+    nrows=1,
+    ncols=2,
+    figsize=("13.5c", "5c"),
+    title="Histogram",
```
Comment (yvonnefroehlich) [inline examples/gallery/histograms/histogram.py:27]: If we keep the changes, there then two histograms 🙂:
```suggestion
    title="Histograms",
```

**Comment by seisman** [comment]
> Maybe a tutorial on histograms beside this gallery example is a solution herer? In a tutorial, we can show the different fill options for the bars, but also how to create cumulative, overlapping, grouped, stacked, horizontal etc. histograms (here as subplot to not post several figues). Maybe we can also state limitations or concerns regarding this kind of visualizations?

A tutorial for histograms is definitely a good idea, although writing a tutorial usually takes more time than writing an example.

**Comment by yvonnefroehlich** [comment]
> > Maybe a tutorial on histograms beside this gallery example is a solution herer? In a tutorial, we can show the different fill options for the bars, but also how to create cumulative, overlapping, grouped, stacked, horizontal etc. histograms (here as subplot to not post several figues). Maybe we can also state limitations or concerns regarding this kind of visualizations?
> 
> A tutorial for histograms is definitely a good idea, although writing a tutorial usually takes more time than writing an example.

I would be pleased to work on such a histogram tutorial. @michaelgrund are you fine with this?

**Review by seisman — APPROVED** [review]

**Comment by yvonnefroehlich** [comment]
> > > Maybe a tutorial on histograms beside this gallery example is a solution herer? In a tutorial, we can show the different fill options for the bars, but also how to create cumulative, overlapping, grouped, stacked, horizontal etc. histograms (here as subplot to not post several figues). Maybe we can also state limitations or concerns regarding this kind of visualizations?
> > 
> > 
> > A tutorial for histograms is definitely a good idea, although writing a tutorial usually takes more time than writing an example.
> 
> I would be pleased to work on such a histogram tutorial. @michaelgrund are you fine with this?

I started working on a tutorial regarding cartesian histograms in PR #2445.

File: examples/gallery/histograms/histogram.py [hunk 5]
```diff
@@ -19,19 +20,39 @@
...
+    ncols=2,
+    figsize=("13.5c", "5c"),
+    title="Histograms",
+):
+    with fig.set_panel(panel=0):
+        fig.histogram(
+            data=data,
+            # Define the frame, add title and set background color to
+            # "lightgray", add labels for x and y axes
+            frame=["WSne+glightgray", "x+lElevation in m", "y+lCounts"],
```
Comment (yvonnefroehlich) [inline examples/gallery/histograms/histogram.py:34]: The main title is now set via the `title` parameter of `pygmt.Figure.subplot`.

```suggestion
            # Define the frame and set background color to "lightgray",
            # add labels for x and y axes
            frame=["WSne+glightgray", "x+lElevation in m", "y+lCounts"],
```

**Comment by weiji14** [comment]
So, what's the decision for this PR? Are we keeping the original 1-histogram example and closing this PR in favour of #2445?

**Comment by michaelgrund** [comment]
Fine with closing this PR.

**Pull request closed without merging**